### PR TITLE
add test for process that exit with non-zero status code

### PR DESCRIPTION
**Pull request by dmartin35**
Signed-off-by: David Martin <david@chaosiq.io>

### CODE REVIEW

File: tests/test_process_provider.py [hunk 1]
```diff
@@ -19,3 +19,13 @@ def test_process_not_utf8_cannot_fail():
     if result['status'] == 0:
         assert result['stderr'] == u''
         assert result['stdout'] == u'é'
+
+
+def test_process_non_exit_zero_warning():
```
Comment (Lawouach) [inline tests/test_process_provider.py]: what does this test assert on?

Comment (dmartin35) [reply tests/test_process_provider.py]: You were right, 
this now asserts the warning message is logged successfully.

File: tests/test_process_provider.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # -*- coding: utf-8 -*-
 import os.path
+from unittest.mock import patch
 from chaoslib.provider.process import run_process_activity
```
Comment (Lawouach) [inline tests/test_process_provider.py]: Need an extra empty line below.
